hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -296,6 +297,12 @@ protected void commit(HoodieTable table, String commitActionType, String instant
     writeTableMetadata(table, instantTime, metadata);
     activeTimeline.saveAsComplete(false, table.getMetaClient().createNewInstant(HoodieInstant.State.INFLIGHT, commitActionType, instantTime),
         serializeCommitMetadata(table.getMetaClient().getCommitMetadataSerDe(), metadata));
+    // update cols to Index as applicable
+    HoodieColStatsIndexUtils.updateColsToIndex(table, config, metadata,
```

Comment:
Can you please remind me why we are updating the columns after writing data commit file? What if we moved this in `writeTableMetadata`?